### DISCUSSION TITLE
‘mergecell’ seems to be abnormal in some cases

### DISCUSSION THREAD

**Discussion by annelhe** [neperfepx/neper#337] [Q&A] [answered]
Hello, I tried to use 'mergecell' to integrate the cells belonging to the same group (divided by volume), but it seems that an error occurs when the volume is changed (the first example can run normally, but the second example will report an error when it runs to the third line ) .
Version 4.2.1-16
Thank you very much.
```text
neper -T -n 100 -domain "cube(10,10,10)" -group "vol<=10?1:2" -statgroup size -statcell group -o test1
neper -T -loadtess test1.tess -transform 'mergecell(group>1)' -o test2
neper -T -loadtess test2.tess -transform 'mergecell(group<2)' -o test3
neper -V test3.tess -datacellcol id -print img

```
```text
neper -T -n 100 -domain "cube(10,10,10)" -group "vol<=8?1:2" -statgroup size -statcell group -o test1
neper -T -loadtess test1.tess -transform 'mergecell(vol<=8)' -o test1
neper -T -loadtess test1.tess -transform 'mergecell(group>1)' -o test2
neper -V teat5.tess -datacellcol id -print img
```
![图片](https://user-images.githubusercontent.com/87170158/143523145-4d9bd2bc-dc06-4734-b55a-15e534e7fe36.png)

**Comment by rquey** [comment, marked as answer]
Cells can be merged only if they are adjacent, which the vol criterion does not guarantee. I think this is your problem in the second case.

**Reply by annelhe** [reply]
Thank you very much for your reply.
